### DISCUSSION TITLE
Should we stop support for Ihatemoney ?

### DISCUSSION THREAD

**Discussion by almet** [spiral-project/ihatemoney#767] [General]
Hi folks,

I've spent some time two months ago to work on the open issues and pull request, asking for feedback from the team, but haven't got any answer. So this makes me wonder if we should "park" the project somehow?

What I have in mind is this : no more new features, just us making updates from time to time so it can continue to work properly.

What do you think about this?

**Comment by Natim** [comment]
I like to see new languages translation on time to time.
With regards to adding new features I like that some people hack around. I'm not sure we should put too much pressure on the timeline. What we could do however is plan a one week-end hackathon once a year to define a roadmap and merge some awaiting PR.

Why do you think the current state of the project is not acceptable?

**Reply by almet** [reply]
Hi, thanks for your answer, it made me think about why I'm leaning this way. The title is a bit strong, but I hope it'll lead to interesting discussions.

> Why do you think the current state of the project is not acceptable?

I took some time a few months back to tackle some of the open pull requests. Some of them were open for about a year. I got almost all of them in good shape and asked for feedback. But then and didn't got any answer.

So, let me be clear : this might seem that I'm renting about the project being un-maintained. It is not what I want to convey here : people have some other priorities in life, and this is — really — fine by me.

But rather than having open pull requests rotting, I would prefer to clarify things, and set expectations. 

Whatever we decide, then let's prepare the potential contributors to how they'll be welcomed.

Now to your proposal of gathering once a year and make progress. I feel this would be great! But lastly my experience contributing to the project was kinda frustrating, so I would be very happy if we find a solution to this.

If it means marking the project as not accepting new features, then why not. That's acceptable by me.

**Comment by Glandos** [comment]
I have to admit that my own expectations were higher than the reality. Basically, it's a psychological effect that hit me: I was glad to throw some involvement into a great project, and it all started fine. Then, I had some other distractions (with various level of seriousness), and diving into the current state became harder every day.

And then, dependabot entered into play (thanks @almet for migrating from dependabot-beta), and triggered a sensitive string in me: we should update our dependencies! It was harder than expected, but I'm glad I did it: Flask 2 and others from Pallet Projects. WTForms and SQLAlchemy is another topic, but the very positive point here is: I took enough time to inspect the project, upgrade my setup and my workflow, so I plan to finally finish #599.

Yes, some requests have been let alone for too long, I don't feel great about it. The main obstacle to me is, as you said, a clear goal of features and support.

Right now, new features can come in, but they should be "ready-to-be-merged", otherwise, we don't have the time to do it ourselves. Regarding support (e.g. #760 ), I personally don't have time to test a setup that is not mine (in this case, gunicorn), but I don't about other people.

**Comment by zorun** [comment]
Thanks for raising this (and thanks for the reviews, was it really 2 months ago already?). Clearly, there has been a general drop in energy in the project since one year. Is it general covid exhaustion? New priorities? Lack of coordination?

In my opinion, the feature level of IHM is overall quite mature and well designed, so it's not shocking that things don't move so much except for maintenance. 

In addition, new features often pose UX challenges (see #603) as well as maintainability challenges, so I'm personally reluctant to add more features or options. I would totally agree with sending a clear message saying _"no more major features for now"_ (where "major" is something like the recent currency and history features).

I agree that the current state is not very satisfying, with an upcoming version 5.0 that has been lingering for more than a year and unresolved issues. I think it's partly because there were lots of new features added at around the same time, and the result is still not polished enough, with only the most difficult problems remaining. And I think it's also partly because the team doesn't have a lot of time / energy. A hackathon would be great, especially because synchronous work is good for interaction and very motivating.

On a personal note, IHM 4.1.5 is currently fullfilling my needs, so I don't have the motivation to do a lot of work on IHM (besides basic maintenance from time to time). And another reason is that my time is more scarce now than last year.

In the short term, let's get IHM 5.0 out soon, even if it's not perfect?

**Comment by Darin755** [comment]
I just discovered this project. Don't take it away!

**Comment by cpebble** [comment]
I'm sorry about the Necro on this discussion. I'd like to chime in here, having used IHM a few times in the past.

There's a popular service in Denmark called "weShare" that just announced they're shutting down as of 01/01/23. A lot of homelabbers/hackers(me included) are thus looking to set up this kind of service. While i'll have to get used to the new feature set, IMO the software is currently "good enough" for daily use; essentially, there's a "want", not a "need" for new features.
Usually that's a good place for OSS to be at, and the pressure for consistent development effort shouldn't be hard.

I would however say that putting up a notice, and potentially splitting your Contributing page into a HACKING and CONTRIBUTING with procedures for development-setup and "procedure for pull requests". This would nudge/encourage people to try to hack on their own issues, and formalize community involvement in source code fixes(aka hopefully take pressure off you as the maintainers).

Just my two :coin: I'll be trying my hand with self-hosting the new docker support :)

**Reply by almet** [reply]
Hi Christian, and thanks for your input. Don't hesitate to propose changes on the contributing page if you think that would make it better suited :-)

**Reply by Glandos** [reply]
Hi, and thanks for the explanation on the "Danish situation".

Contributing to this project isn't harder than any other Python with Web front end project, but… I'm not really impartial on these. However, we had a lot of new contributors during spring and summer 2020 (remember the pandemic ?), so I guess this was not hard.
As Alexis said, if you can contribute to documentation, don't be shy. And if some important documentation is missing, open a ticket or a discussion.

As a final note, IHateMoney clearly lacks some features, but we also agree on the facts that it doesn't need to have **too much** of them.
